app/query/processors.go

```diff
@@ -1,97 +1,122 @@
 package query
 
 import (
-	"encoding/json"
 	"fmt"
 
+	"github.com/lbryio/lbrytv-player/pkg/paid"
 	"github.com/lbryio/lbrytv/config"
-	"github.com/lbryio/lbrytv/internal/errors"
 
 	ljsonrpc "github.com/lbryio/lbry.go/v2/extras/jsonrpc"
 
-	log "github.com/sirupsen/logrus"
 	"github.com/ybbus/jsonrpc"
 )
 
-func postProcessResponse(response *jsonrpc.RPCResponse, query *jsonrpc.RPCRequest) error {
-	switch query.Method {
+type PurchaseReceipt struct {
```

Comment:
no need to make this exported. I always err on the side of leaving everything unexported. Keeps the package's API small. If you need to export it later, its very easy. Going the other way is harder.

Reply:
As someone who did Python for too long, this feels a bit foreign. Is there a good reason to err on the side of privateness? 

I remember mentioning in another discussion that I was inconvenienced a few times by stumbling into variables that were private. But I don't recall ever saying to myself: damn, I wish this `type` was made private by the package author.

Reply:
here's how i see it: think of every Go package (including all the sub-packages in lbrytv) as it's own library, and every exported thing as part of that library's API. the API is a promise to other packages. if you want to add to the API, it's easy. if you want to change the API or drop things, you're now breaking your promise to any package that uses your package.

with those incentives, you want to promise as little as possible. if another package needs something from you, you can export an unexported object to add to your API. if you export too much, you'll have to consider more possibilities when you do want to change something. 

related: only exporting what's useful to other packages reduces the cognitive load on those packages. pretend i'm an external developer. if i see `query.PurchaseReceipt` exported, that's an [affordance](https://www.interaction-design.org/literature/topics/affordances). i may say "oh cool, a thing i can use for something. what does it do?". but in reality there's nothing for me to do with it. its not useful to me.

this is related to the logic behind deleting code that's not used, and also related to the principle of least privilege (though not in a security sense).

Reply:
speaking of Python, I don't like "private-by-convention" at all. i get the logic for it and the appeal. treating other devs as "responsible adults who play nice together" is reasonable. it also increases the range of possibilities and opens the door to mistakes. 

for more on this, i highly recommend Rich Hickey's [Simple Made Easy](https://github.com/matthiasn/talk-transcripts/blob/master/Hickey_Rich/SimpleMadeEasy.md). his idea of complecting as the opposite of simplicity is very influential to me